tests/ui/fmt/issue-116631.rs

```diff
@@ -0,0 +1,3 @@
+fn main() {
+    format_args!("{}\n", 0xffff_ffff_u8); //~ ERROR literal out of range for `u8`
```

Comment:
Should this have a test for each of the different integer types?

Reply:
At the very least it should have a test case that covers #115423 (which I don't interpet is about being out of range, but rather about not respecting the type of the literal).